src/couch/src/couch_db.erl

```diff
@@ -1589,7 +1476,7 @@ open_doc_int(Db, #full_doc_info{id=Id,rev_tree=RevTree}=FullDocInfo, Options) ->
         {ok, Doc#doc{meta=doc_meta_info(DocInfo, RevTree, Options)}}, Options);
 open_doc_int(Db, Id, Options) ->
     case get_full_doc_info(Db, Id) of
-    {ok, FullDocInfo} ->
+    #full_doc_info{} = FullDocInfo ->
```

Comment:
I assume you're thinking in terms of performance? AFAIK matching records is fairly well optimized so I'm not too worried there. Also we're basically doing the same operation as matching {ok, any()} so its not any slower than the old way.

I also don't think of this as defensive so much as assertive. Defensive to me would be if we added an Error clause that was then thrown. Here was just say "these two things may come back and then vomit if we see anything else". And using the record match tells readers what we're getting back there.